### ISSUE TITLE
compilation with "make NOASM=1" fails

### ISSUE THREAD

**Issue by hannob** [veracrypt/VeraCrypt#563]
For some tests I wanted to disable asm optimizations, however compilation with "make NOASM=1" fails (current git code):
```
Compiling cpu.c
In file included from ../Crypto/cpu.h:196,
                 from ../Crypto/cpu.c:3:
../Crypto/cpu.c: In function ‘TryAESNI’:
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/include/wmmintrin.h:61:1: error: inlining failed in call to always_inline ‘_mm_aesenc_si128’: target specific option mismatch
   61 | _mm_aesenc_si128 (__m128i __X, __m128i __Y)
      | ^~~~~~~~~~~~~~~~
../Crypto/cpu.c:257:14: note: called from here
  257 |   ciphered = _mm_aesenc_si128(block, subkey);
      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../Crypto/cpu.h:196,
                 from ../Crypto/cpu.c:3:
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/include/wmmintrin.h:61:1: error: inlining failed in call to always_inline ‘_mm_aesenc_si128’: target specific option mismatch
   61 | _mm_aesenc_si128 (__m128i __X, __m128i __Y)
      | ^~~~~~~~~~~~~~~~
../Crypto/cpu.c:257:14: note: called from here
  257 |   ciphered = _mm_aesenc_si128(block, subkey);
      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```

I wondered if I also have to disable sse2, but then I just get a different error:
```
In file included from ../Crypto/Camellia.c:48:
/tmp/VeraCrypt/src/Common/Crypto.h:125:2: error: unknown type name ‘wchar_t’
  125 |  wchar_t *Name;   // Name
      |  ^~~~~~~
/tmp/VeraCrypt/src/Common/Crypto.h:146:2: error: unknown type name ‘wchar_t’
  146 |  wchar_t *Name;    // Name
      |  ^~~~~~~
In file included from ../Crypto/Camellia.c:48:
/tmp/VeraCrypt/src/Common/Crypto.h:324:7: error: unknown type name ‘wchar_t’
  324 | const wchar_t * CipherGetName (int cipher);
      |       ^~~~~~~
/tmp/VeraCrypt/src/Common/Crypto.h:345:1: error: unknown type name ‘wchar_t’
  345 | wchar_t * EAGetName (wchar_t *buf, int ea, int guiDisplay);
      | ^~~~~~~
In file included from ../Crypto/Camellia.c:48:
/tmp/VeraCrypt/src/Common/Crypto.h:227:1: note: ‘wchar_t’ is defined in header ‘<stddef.h>’; did you forget to ‘#include <stddef.h>’?
  226 | #include "config.h"
  +++ |+#include <stddef.h>
  227 | 
In file included from ../Crypto/Camellia.c:48:
/tmp/VeraCrypt/src/Common/Crypto.h:345:22: error: unknown type name ‘wchar_t’
  345 | wchar_t * EAGetName (wchar_t *buf, int ea, int guiDisplay);
      |                      ^~~~~~~
/tmp/VeraCrypt/src/Common/Crypto.h:345:22: note: ‘wchar_t’ is defined in header ‘<stddef.h>’; did you forget to ‘#include <stddef.h>’?
/tmp/VeraCrypt/src/Common/Crypto.h:346:18: error: unknown type name ‘wchar_t’
  346 | int EAGetByName (wchar_t *name);
      |                  ^~~~~~~
/tmp/VeraCrypt/src/Common/Crypto.h:346:18: note: ‘wchar_t’ is defined in header ‘<stddef.h>’; did you forget to ‘#include <stddef.h>’?
/tmp/VeraCrypt/src/Common/Crypto.h:352:1: error: unknown type name ‘wchar_t’
  352 | wchar_t * EAGetModeName (int ea, int mode, BOOL capitalLetters);
      | ^~~~~~~
/tmp/VeraCrypt/src/Common/Crypto.h:352:1: note: ‘wchar_t’ is defined in header ‘<stddef.h>’; did you forget to ‘#include <stddef.h>’?
/tmp/VeraCrypt/src/Common/Crypto.h:371:7: error: unknown type name ‘wchar_t’
  371 | const wchar_t *HashGetName (int hash_algo_id);
      |       ^~~~~~~
/tmp/VeraCrypt/src/Common/Crypto.h:376:20: error: unknown type name ‘wchar_t’
  376 | void HashGetName2 (wchar_t *buf, int hashId);
      |                    ^~~~~~~
/tmp/VeraCrypt/src/Common/Crypto.h:376:20: note: ‘wchar_t’ is defined in header ‘<stddef.h>’; did you forget to ‘#include <stddef.h>’?
```

**Comment by alt3r-3go** [comment]
I'm looking into this and I think have a fix, will summarize it later today, hopefully.

**Comment by alt3r-3go** [comment]
@hannob, okay, so I've got the proposed fix in #568 - feel free to test out. At least it does compile now :smiley: 

@idrassi, as mentioned in the PR, if you think the approach is incorrect or I missed anything - please just let me know, I'll correct.
